app/models/megaphone/episode.rb

```diff
@@ -145,8 +145,9 @@ def delete_delivery_status
     # like when dtr wasn't ready at first
     # so we can make that update and then mark uploaded
     def upload_audio!
-      set_audio_attributes
-      update!
+      # basically just do an update, passing in the episode
+      # which will override the audio attributes
+      update!(feeder_episode)
```

Comment:
ok, when this is called, it does a retrieve of the current attributes on megaphone, then writes those back - but, for the summary, each cycle of doing that adds the ad choices line at the end, so after a few audio saves, it ends up repeated a bunch of times. Instead, go with the current attributes for audio and everything else when updating.